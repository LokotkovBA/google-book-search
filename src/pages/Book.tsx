import React from 'react';
import { useLocation } from 'react-router';
import '../styles/Book.scss';

const Book: React.FC = () => {
    const { state } = useLocation();

    return (
        <article data-testid="book-page" className="book-page">
            {state?.cover_src ? <img className="cover book-page__cover" width={233} height={350} alt="Book cover" src={state?.cover_src} /> : <div className="book-page__cover book-page__placeholder--cover" />}
            <article className="book-details">
                {state?.categories ?
                    <p className="book-details__categories book-details__text">
                        {state.categories.join(' / ')}
                    </p> : <p className="book-details__categories book-details__text placeholder" />}
                {state?.title ?
                    <h2 className="book-details__title book-details__text">
                        {state.title}
                    </h2> : <p className="book-details__title book-details__text placeholder" />}
                {state?.authors ? <p className="book-details__authors book-details__text">
                    {state.authors.map((author: string, index: number, array: string[]) => index !== array.length - 1 ? `${author}, ` : author)}
                </p> : <p className="book-details__authors book-details__text placeholder" />}
                {state?.description ?
                    <p className="book-details__description book-details__text">
                        {state.description}
                    </p> : <p className="book-details__description book-details__text placeholder" />}
            </article>
        </article>);
};

export default Book;
