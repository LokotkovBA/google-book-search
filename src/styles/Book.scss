.book-page {
    background-color: var(--card-color);
    border-radius: 2em;
    margin-top: 2em;
    display: grid;
    grid-template-columns: .9fr 1fr;
    justify-items: center;
    align-items: start;

    &__cover {
        margin: 3em;
    }

    &__placeholder {
        &--cover {
            width: 233px;
            height: 350px;
        }
    }
}

.book-details {
    border-radius: 0 2em 2em 0;
    padding: 2em;
    background-color: var(--search-bar-color);
    min-width: 100%;
    max-width: 60ch;
    height: 100%;
    text-align: left;

    &__text {
        padding: 0 1.5rem;
    }

    &__description {
        border-radius: 2em;
        padding: 1.5rem;
        background-color: var(--background-color);
        min-height: 10ch;
    }
}


@media screen and (width < 56.3rem) {
    .book-page {
        grid-template-columns: 1fr;

        &__cover {
            margin: 3em 0;
        }
    }

    .book-details {
        border-radius: 0 0 2em 2em;
        padding: 1em;
    }
}