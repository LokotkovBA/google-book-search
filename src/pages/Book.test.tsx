import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import 'react-router-dom';
import Book from './Book';

const ComponentToTest = () => {
    return (
        <MemoryRouter initialEntries={['/books/1']}>
            <Book />
        </MemoryRouter>);
};


describe('Book', () => {
    it('should render empty book', () => {
        render(<ComponentToTest />);
        expect(screen.getByTestId('book-page')).toMatchSnapshot();
    });
});
