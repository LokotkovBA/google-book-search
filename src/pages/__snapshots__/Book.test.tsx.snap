// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book should render empty book 1`] = `
<article
  class="book-page"
  data-testid="book-page"
>
  <div
    class="book-page__cover book-page__placeholder--cover"
  />
  <article
    class="book-details"
  >
    <p
      class="book-details__categories book-details__text placeholder"
    />
    <p
      class="book-details__title book-details__text placeholder"
    />
    <p
      class="book-details__authors book-details__text placeholder"
    />
    <p
      class="book-details__description book-details__text placeholder"
    />
  </article>
</article>
`;
